src/main/java/com/minecolonies/coremod/quests/objectives/BreakBlockObjective.java

```diff
@@ -0,0 +1,167 @@
+        this.nextObjective = nextObjective;
+        this.blockToMine = blockToMine;
+    }
+
+    /**
+     * Parse the mine block objective from json.
+     * @param jsonObject the json to parse it from.
+     * @return a new objective object.
+     */
+    public static IQuestObjective createObjective(final JsonObject jsonObject)
+    {
+        JsonObject details = jsonObject.getAsJsonObject(DETAILS_KEY);
+        final int target = details.get(TARGET_KEY).getAsInt();
+        final int quantity = details.get(QUANTITY_KEY).getAsInt();
+        final Block block = ForgeRegistries.BLOCKS.getHolder(new ResourceLocation(details.get(BLOCK_KEY).getAsString())).get().get();
+        final int nextObj = details.has(NEXT_OBJ_KEY) ? details.get(NEXT_OBJ_KEY).getAsInt() : -1;
+
+        return new BreakBlockObjective(target, quantity, block, nextObj, parseRewards(jsonObject));
+    }
+
+    @Override
+    public IObjectiveData init(final IColonyQuest colonyQuest)
+    {
+        super.init(colonyQuest);
+        if (colonyQuest.getColony() instanceof Colony)
+        {
+            // Only serverside cleanup.
+            ((Colony) colonyQuest.getColony()).getEventHandler().addQuestObjectiveListener(this.blockToMine, colonyQuest.getAssignedPlayer(), colonyQuest);
+        }
+        return new BlockMiningProgressData();
+    }
+
+    @Nullable
+    @Override
+    public IObjectiveData getObjectiveData()
+    {
+        return new BlockMiningProgressData();
+    }
+
+    @Override
+    public void onAbort(final IColonyQuest colonyQuest)
+    {
+        cleanupEvent(colonyQuest);
+    }
+
+    private void cleanupEvent(final IColonyQuest colonyQuest)
+    {
+        if (colonyQuest.getColony() instanceof Colony)
+        {
+            // Only serverside cleanup.
+            ((Colony) colonyQuest.getColony()).getEventHandler().removeQuestObjectiveListener(this.blockToMine, colonyQuest.getAssignedPlayer(), colonyQuest);
+        }
+    }
+
+    @Override
+    public void onBlockBreak(final IObjectiveData blockMiningProgressData, final IColonyQuest colonyQuest, final Player player)
+    {
+        if (blockMiningProgressData.isFulfilled())
+        {
+            return;
+        }
+
+        ((BlockMiningProgressData) blockMiningProgressData).currentProgress++;
+        if (blockMiningProgressData.isFulfilled())
+        {
+            colonyQuest.advanceObjective(player, nextObjective);
+        }
+    }
+
+    @Override
+    public void onWorldLoad(final IColonyQuest colonyQuest)
+    {
+        super.onWorldLoad(colonyQuest);
+        if (colonyQuest.getColony() instanceof Colony)
+        {
+            // Only serverside cleanup.
+            ((Colony) colonyQuest.getColony()).getEventHandler().addQuestObjectiveListener(this.blockToMine, colonyQuest.getAssignedPlayer(), colonyQuest);
+        }
+    }
+
+    /**
+     * Progress data of this objective.
+     */
+    public class BlockMiningProgressData implements IObjectiveData
```

Comment:
Maybe move those out of the type classes, it is kinda confusing having Type and Instance classes in the same file

Reply:
Didnt see this as an instance, just as a small tracking container. Not all Objectives have and need it.